LEAF_Request_Portal/templates/reports/sitemaps_template.tpl

```diff
@@ -1,32 +1,180 @@
+    function updateCardUI(cardID){
+        $.each(sitemapOBJ.cards, function(index, value){
+            if(value.id == cardID){
+    	$('#li_cardID_'+cardID+' a').text(value.title);
+        $('#div_cardID_'+cardID+' h3').text(value.title);
+        $('#div_cardID_'+cardID+' p').text(value.description);
+            }
+        });
+    }
+    
+	//generate unique id for sitemap card
+    function generateNewCardID(){
+        do {
+           var result           = '';
+           var characters       = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789';
+           for ( var i = 0; i < 5; i++ ) {
+              result += characters.charAt(Math.floor(Math.random() * 62));
+           }
+        }
+        while (cardIDExists(result));
+        return result;
+    }
+    
+	//check if unique id already exists
+    function cardIDExists(newID) {
+        $.each(sitemapOBJ.cards,  function(index, value){
+        	if(value.id == newID){
+            	return true;
+            }
+        });
+        return false;
+    }
+    
+	//brings up dialog to add a card
     function createGroup() {
+        var dialog = createNewCardDialog();
         dialog.setTitle('Add New Card');
-        dialog.setContent('<div><div role="heading">Card Title: </div><input aria-label="" id=""></input><div role="heading" style="margin-top: 1rem;">Card Description: </div><input aria-label="Enter group name" id=""></input><div role="heading" style="margin-top: 1rem;">Target Site Address: </div><input aria-label="" id=""></input></div>');
+        dialog.setContent('<div><div role="heading">Card Title: </div><input aria-label="" id="card-title"></input><div role="heading" style="margin-top: 1rem;">Card Description: </div><input aria-label="Enter group name" id="card-description"></input><div role="heading" style="margin-top: 1rem;">Target Site Address: </div><input aria-label="" id="card-target"></input></div>');
 
         dialog.show();
         $('input:visible:first, select:visible:first').focus();
     }
 
-    var dialog;
-    $(function() {
-	    dialog = new dialogController('xhrDialog', 'xhr', 'loadIndicator', 'button_save', 'button_cancelchange');
-
+	//instantiates new card dialog
+    function createNewCardDialog() {
+            var dialog = new dialogController('xhrDialog', 'xhr', 'loadIndicator', 'button_save', 'button_cancelchange');
+       	 	dialog.setSaveHandler(function() {
+            dialog.indicateBusy();
+            var id = generateNewCardID();
+            var title = $("#xhr input#card-title").val();
+            var description = $("#xhr input#card-description").val();
+            var target = $("#xhr input#card-target").val();
+            var order = sitemapOBJ.cards.length;
+            var newCard = {id: id, title: title, description: description, target: target, order: order};
+            sitemapOBJ.cards.push(newCard);
+            addCardToUI(newCard);
+            dialog.hide();
```

Comment:
```show()``` hasn't been called yet, is ```hide()``` needed?

Reply:
this is in the saveHandler() for the dialog, so this would only get called after show()